src/app/pages/send/send-crypto-asset-form/form/btc/use-btc-send-form.tsx

```diff
@@ -13,31 +15,47 @@ import {
   btcAddressNetworkValidator,
   btcAddressValidator,
 } from '@app/common/validation/forms/address-validators';
-import { btcAmountPrecisionValidator } from '@app/common/validation/forms/currency-validators';
-import { useBitcoinCryptoCurrencyAssetBalance } from '@app/query/bitcoin/address/address.hooks';
+import {
+  btcAmountPrecisionValidator,
+  btcInsufficientBalanceValidator,
+} from '@app/common/validation/forms/currency-validators';
+import { useBitcoinAssetBalance } from '@app/query/bitcoin/address/address.hooks';
 import { useCurrentBtcAccountAddressIndexZero } from '@app/store/accounts/blockchain/bitcoin/bitcoin-account.hooks';
 import { useCurrentNetwork } from '@app/store/networks/networks.selectors';
 
+import { useCalculateMaxBitcoinSpend } from '../../family/bitcoin/hooks/use-calculate-max-spend';
 import { useSendFormNavigate } from '../../hooks/use-send-form-navigate';
 import { useGenerateSignedBitcoinTx } from './use-generate-bitcoin-raw-tx';
 
 export function useBtcSendForm() {
+  const formRef = useRef<FormikProps<BitcoinSendFormValues>>(null);
+
   const currentNetwork = useCurrentNetwork();
   const currentAccountBtcAddress = useCurrentBtcAccountAddressIndexZero();
-  const btcCryptoCurrencyAssetBalance =
-    useBitcoinCryptoCurrencyAssetBalance(currentAccountBtcAddress);
+  const btcCryptoCurrencyAssetBalance = useBitcoinAssetBalance(currentAccountBtcAddress);
   const { isShowingHighFeeConfirmation, setIsShowingHighFeeConfirmation } = useDrawers();
   const { whenWallet } = useWalletType();
   const sendFormNavigate = useSendFormNavigate();
   const generateTx = useGenerateSignedBitcoinTx();
+  const calcMaxSpend = useCalculateMaxBitcoinSpend();
 
   return {
+    formRef,
+
     currentNetwork,
 
     validationSchema: yup.object({
-      amount: btcAmountPrecisionValidator(
-        formatPrecisionError(btcCryptoCurrencyAssetBalance.balance)
-      ),
+      amount: yup
+        .number()
+        .concat(
+          btcAmountPrecisionValidator(formatPrecisionError(btcCryptoCurrencyAssetBalance.balance))
+        )
+        .concat(
+          btcInsufficientBalanceValidator({
+            recipient: formRef.current?.values.recipient ?? '',
```

Comment:
Seems the purpose of `formRef` is to access the value of another field during validation? Yup has some support for validating related fields (e.g., [using context `this`](https://github.com/jquense/yup#schematestname-string-message-string--function--any-test-function-schema)), although perhaps using Yup's approach would complicate things?

Reply:
Good feedback, will see if I can do it this way.

Reply:
Comment here. Agree this would be better but can do later